django/middleware/security.py

```diff
@@ -52,4 +53,13 @@ def process_response(self, request, response):
                 if isinstance(self.referrer_policy, str) else self.referrer_policy
             ))
 
+        if self.feature_policy:
+            response.setdefault('Feature-Policy', '; '.join(
+                '%s %s' % (directive, ' '.join(
+                    ('%r' if i in {'none', 'self'} else '%s') % i for i in (
+                        allowlist.split(' ') if isinstance(allowlist, str) else allowlist
```

Comment:
This allows the allow-list dict value to be either a string or a list of strings. IMO it will be better to only support one method (insert zen of Python reference here).

I think only supporting the string would be better, because then the quoting of `self` etc. is explicit and there are less surprises.

Reply:
We allowed supporting both for `Referrer-Policy`, but then that is a much simpler header and I agree it makes things a bit too complex here.

> I think only supporting the string would be better, because then the quoting of self etc. is explicit and there are less surprises.

So I would err the other way otherwise we would need to properly parse this rather than be provided with a simple list of strings for each feature.

Do you have an opinion on this one @carltongibson?